public/app/features/dashboard/routes.ts

```diff
@@ -0,0 +1,24 @@
+import { SafeDynamicImport } from '../../core/components/DynamicImports/SafeDynamicImport';
+import { config } from '../../core/config';
+import { RouteDescriptor } from '../../core/navigation/types';
+import { DashboardRoutes } from '../../types';
+
+export const getPublicDashboardRoutes = (): RouteDescriptor[] => {
+  if (config.featureToggles.publicDashboards) {
+    return [
+      {
+        path: '/p/:uid/:slug?',
```

Comment:
I would have expected this to include a key as well, and be a route like 
`/public-dashboard/:key/:slug?`  

I don't think we should mix the concepts of uid with public dashboard access keys, because many access keys will be able lead to the same dashboard. I would think that this uid is the dashboard uid that is shared, but I assume it's not.  

I had a similar question here: https://github.com/grafana/grafana/pull/49131 , what the uid for a public dashboard was for, is it the same as a key? but a single public dashboard can be shared to multiple emails (each with their own key), so that won't scale. But maybe it's just the public dashboard config id for now and worry about access keys later?

Reply:
Yeah this url is going to change. This is just temporary because the backend work is in progress